File: leetcode-js/26.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    let currIdx = 0;
    let currNum = nums[0];
    for (let num of nums) {
        if (num != currNum) {
            currNum = num;
            currIdx++;
            nums[currIdx] = num;
        }
    }
    return currIdx + 1;
};

const nums = [1, 1, 1, 1, 2, 2, 2, 2, 3, 4, 4, 5, 5, 6, 6];
console.log(removeDuplicates(nums));

File: leetcode-go/26.go
package main

func removeDuplicates(nums []int) int {
	currNum := nums[0]
	currIdx := 0
	for _, num := range nums {
		if num != currNum {
			currNum = num
			currIdx++
			nums[currIdx] = num
		}
	}
	return currIdx + 1
}
